google/cloud/spanner_dbapi/exceptions.py

```diff
@@ -27,7 +29,65 @@ class Error(Exception):
     Does not include :class:`Warning`.
     """
 
-    pass
+    def _is_error_cause_instance_of_google_api_exception(self):
+        return isinstance(self.__cause__, GoogleAPICallError)
+
+    @property
+    def reason(self):
+        """The reason of the error.
+        Reference:
+            https://github.com/googleapis/googleapis/blob/master/google/rpc/error_details.proto#L112
```

Comment:
We normally don't refer to github proto links in reference. If there are any google documentation pages, you can add those instead.

Reply:
Yes, we can replace it with following document: 
https://cloud.google.com/apis/design/errors#error_info